File: hw/jacome/U3/HW22Strategy/JS/model/InsertionSort.js
const SortingStrategy = require('./SortingStrategy');

class InsertionSort extends SortingStrategy {
    sort(data) {
        const n = data.length;
        for (let i = 1; i < n; ++i) {
            const key = data[i];
            let j = i - 1;
            while (j >= 0 && data[j] > key) {
                data[j + 1] = data[j];
                j = j - 1;
            }
            data[j + 1] = key;
        }
        return data;
    }
}

module.exports = InsertionSort;

File: hw/paillacho/U3/HW22Strategy/StrategyPython/model/insertion_sort.py
from .sorting_strategy import SortingStrategy

# Author: Carlos Paillacho


class InsertionSort(SortingStrategy):

    def sort(self, data):
        n = len(data)
        for i in range(1, n):
            key = data[i]
            j = i - 1
            while j >= 0 and data[j] > key:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = key
        return data
